src/op_mode/show_nat_rules.py

```diff
@@ -40,6 +40,16 @@
     data_json = jmespath.search('nftables[?rule].rule[?chain]', tmp)
     for idx in range(0, len(data_json)):
         data = data_json[idx]
+        
+        # If there is no index 3, we don't think this is the record we need to check
```

Comment:
Band aid fixes have their place, but they need to be properly explained. **Why** can rules be invalid and not have required fields?

Can we prevent such rules from being created? If not, can we add special display rules for them? Do you have examples of such rules from real-life setups?

Reply:
The invalid rule here refers to the rule that we should not process and print out in the operation mode. The nft rule is as follows

```
table ip nat {
        chain PREROUTING {
                type nat hook prerouting priority dstnat; policy accept;
                iifname "pppoe0" counter packets 1146798 bytes 104964883 jump MINIUPNPD
                iifname "pppoe0" counter packets 1146703 bytes 104955703 jump MINIUPNPD
        }

        chain INPUT {
                type nat hook input priority 100; policy accept;
        }

        chain POSTROUTING {
                type nat hook postrouting priority srcnat; policy accept;
                oifname "pppoe0" ip saddr 192.168.0.0/16 counter packets 4104891 bytes 357568379 log prefix "[NAT-SRC-1-MASQ]" comment "SRC-NAT-1"
                oifname "pppoe0" ip saddr 192.168.0.0/16 counter packets 4104891 bytes 357568379 masquerade comment "SRC-NAT-1"
        }

        chain OUTPUT {
                type nat hook output priority -100; policy accept;
        }

        chain VYATTA_PRE_DNAT_HOOK {
        }

        chain VYATTA_PRE_SNAT_HOOK {
        }

        chain MINIUPNPD {
        }
}

```

Therefore, when the rule field is less than the expected field, the processing of the program will report an error for exceeding the index, so we need to filter it

In order to strengthen the verification, I also checked several fields and only printed specific rules